src/shouldHideEvent.js

```diff
@@ -14,38 +14,37 @@
  limitations under the License.
  */
 
-function _isLeaveOrJoin(ev) {
-    const isMemberEvent = ev.getType() === 'm.room.member' && ev.getStateKey() !== undefined;
-    if (!isMemberEvent) {
-        return false; // bail early: all the checks below concern member events only
-    }
+function memberEventDiff(ev) {
+    const diff = {
+        // a Member Event is a State Event and so its State Key must not be undefined.
+        isMemberEvent: ev.getType() === 'm.room.member' && ev.getStateKey() !== undefined,
```

Comment:
Currently, `isMembershipChange` in `MessagePanel` only checks the type so I think it should be fine.